include/ardpolyfills/arduino.h

```diff
@@ -21,14 +21,15 @@
 #include <cctype>
 #include <cmath>
 #include <cstdlib>
+#include <algorithm>
 
 // Arduino Math functions
 using std::abs;
 long constrain(long, long, long);
 long map(long, long, long, long, long);
-template <typename T> T min(T, T);
-template <typename T> T max(T, T);
-template <typename T> T sq(T);
+using std::max;
+using std::min;
+template <typename T> T sq(T) { return (x * x); };
```

Comment:
If I was to compile this, I'd be greeted by something along the lines of:
"Fatal error: undeclared name `x`"

Reply:
it is a bit weird that the VS doesn't tell you that.